transport/protocol/src/ack/codec.rs

```diff
@@ -0,0 +1,177 @@
+use std::marker::PhantomData;
```

Comment:
This codec looks general purpose, should it be moved outside of the ack protocol?

Reply:
That's the code cleanup part. There will only be one codec and this CBOR based will be removed. But we needed something temporary there and I implemented this before we agreed on unifying the payload content.

Reply:
ok